### PR TITLE
Add se3.left_act()

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by mhmukadam — APPROVED** [review]

File: theseus/geometry/functional/se3.py [hunk 1]
```diff
@@ -49,6 +49,11 @@ def check_project_matrix(matrix: torch.Tensor):
     return matrix.shape[-2:] == (3, 4)
 
 
+def check_left_act_matrix(matrix: torch.Tensor):
+    if matrix.shape[-2] != 3:
+        raise ValueError("Inconsistent shape for the matrix.")
```
Comment (luisenp) [inline theseus/geometry/functional/se3.py:54]: Add more explicit error message.

File: theseus/geometry/functional/se3.py [hunk 2]
```diff
@@ -653,3 +658,45 @@ def backward(cls, ctx, grad_output):
...
+
+def _left_act_backward_helper(group, matrix, grad_output) -> torch.Tensor:
+    jac_rot = torch.einsum("n...ij,n...kj->n...ik", grad_output, matrix)
+    if matrix.ndim > 3:
+        dims = list(range(1, matrix.ndim - 2))
+        jac_rot = jac_rot.sum(dims)
+    return torch.cat((jac_rot, jac_rot.new_zeros(jac_rot.shape[0], 3, 1)), dim=-1)
+
+
+class LeftAct(lie_group.BinaryOperator):
```
Comment (luisenp) [inline theseus/geometry/functional/se3.py:682]: Missing description of what this does (or reference).

File: theseus/geometry/functional/se3.py [hunk 3]
```diff
@@ -653,3 +658,45 @@ def backward(cls, ctx, grad_output):
...
+# -----------------------------------------------------------------------------
+def _left_act_impl(group: torch.Tensor, matrix: torch.Tensor) -> torch.Tensor:
+    if not check_group_tensor(group):
+        raise ValueError("Invalid data tensor for SE3.")
+    check_left_act_matrix(matrix)
+    ret = so3._left_act_impl(group[:, :, :3], matrix)
+    return ret
+
+
+def _left_act_backward_helper(group, matrix, grad_output) -> torch.Tensor:
```
Comment (luisenp) [inline theseus/geometry/functional/se3.py:674]: Missing type information for input arguments.

**Review by luisenp — CHANGES_REQUESTED** [review]
LGTM. Some small comments to address.

**Pull request merged**